ts/src/handlers.ts

```diff
@@ -349,6 +350,20 @@ export class Handlers {
                 throw utils.getInvalidFunctionCallError(decodedCalldata.functionName);
         }
     }
+    public async postSoftCancelsAsync(req: express.Request, res: express.Response): Promise<void> {
+        utils.validateSchema(req.body, softCancelsSchema);
+        
+        const softCancelsFound = await orderModel.findSoftCancelledOrdersByHashAsync(req.body.orderHashes);
+
+        if (!softCancelsFound.length) {
+            res.status(HttpStatus.NOT_FOUND).send('NOT_FOUND');
```

Comment:
@Arctek this interface feels a bit off to me. No soft cancels being found is a valid response. The request did perform the query, and it did return an empty set, which is not an exceptional circumstance. Therefore the API should return an empty array in this case, rather then returning a 404.

More context on this line of reasoning from a StackOverflow post:

When in doubt, consult the documentation. Reviewing the W3C definitions for HTTP Status codes, gives us this:

    200 OK - The request has succeeded. The information returned with the response is dependent on the method used in the request.

    404 Not Found - The server has not found anything matching the Request-URI.

In the context of your API, it very much depends on how queries are created and how objects are retrieved. But, my interpretation has always been that:

    If I ask for a particular object, and it exists return 200 code, if it doesn't exist return the correct 404 code.
    But, if I ask for a set of objects that match a query, a null set is a valid response and I want that returned with a 200 code. The rationale for this is that the query was valid, it succeeded and the query returned nothing.